libs/wingsdk/src/http/http.ts

```diff
@@ -271,6 +327,61 @@ export class Util {
     });
   }
 
+  /**
+   * Parses the input URL String using WHATWG URL API and returns an URL Struct.
+   * @param urlString The URL String to be parsed.
+   * @throws Will throw an error if the input String is not a valid URL.
+   * @inflight
+   * @returns An URL Struct.
+   */
+  public static parseUrl(urlString: string): Url {
+    try {
+      const nodeUrl = new NodeUrl(urlString);
+      return {
+        href: nodeUrl.href,
+        protocol: nodeUrl.protocol,
+        host: nodeUrl.host,
+        hostname: nodeUrl.hostname,
+        port: nodeUrl.port,
+        pathname: nodeUrl.pathname,
+        search: nodeUrl.search,
+        hash: nodeUrl.hash,
+        origin: nodeUrl.origin,
+        username: nodeUrl.username,
+        password: nodeUrl.password,
+      };
+    } catch (error) {
+      throw new Error(`Invalid URL: ${urlString}`);
```

Comment:
Is there useful info in `error` that we want to include in the message?

Reply:
No,` error.message` is actually `Invalid URL` so I could have wrote `${error.message}: ${urlString}` and it would have been equivalent but less readable from the code I think.